components/cloud_usage_modal/workspace_limits_panel.tsx

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {FormattedMessage, useIntl} from 'react-intl';
+
+import {Limits} from '@mattermost/types/cloud';
+import {CloudUsage} from 'components/common/hooks/useGetUsage';
+import UsagePercentBar from 'components/common/usage_percent_bar';
+import {asGBString, inK} from 'utils/limits';
+
+import './workspace_limits_panel.scss';
+
+export interface Message {
+    id: string;
+    defaultMessage: string;
+    values?: Record<string, any>;
+}
+
+export function messageToElement(x: Message | React.ReactNode): React.ReactNode {
+    if (Object.prototype.hasOwnProperty.call(x, 'defaultMessage')) {
+        return (
+            <FormattedMessage
+                id={(x as Message).id}
+                defaultMessage={(x as Message).defaultMessage}
+                values={(x as Message).values}
+            />
+        );
+    }
+
+    return x;
+}
+
+interface LimitLineProps {
+    icon: string;
+    limitName: React.ReactNode;
+    limitStatus: React.ReactNode;
+    percent: number;
+    showIcons?: boolean;
+}
+
+function LimitLine(props: LimitLineProps) {
```

Comment:
shouldn't it be better place this in its own file?

Reply:
yep! will do.